### PR TITLE
Use `action.argv` for CC targets as well

### PR DESCRIPTION
**Pull request by brentleyjones**
This will generally make our compiler flags match better with reality. This is also an analysis phase wall-time reduction. Though it’s mainly a change that will let us use `action.args` for copt processing in the future, reducing our memory footprint.

### CODE REVIEW

**Comment by brentleyjones** [comment]
@MobileNativeFoundation/rules_xcodeproj-maintainers I've checked the difference in the example `{c,cxx}.compile.params` files, but I would like you to compare on your end as well to make sure nothing odd is coming through now, which could break indexing.

**Comment by brentleyjones** [comment]
Looks like if we take this change (which I plan on doing) we are dropping Bazel 5 support.

**Comment by jpsim** [comment]
> Looks like if we take this change (which I plan on doing) we are dropping Bazel 5 support.

Is it worth doing one last release prior to dropping Bazel 5? 1.4.1?

**Comment by brentleyjones** [comment]
Yeah, I won't merge this for a bit knowing it breaks compatibility. And we will release 1.4.1 or 1.5.0 before the drop.

**Comment by thiagohmcruz** [comment]
> Is it worth doing one last release prior to dropping Bazel 5? 1.4.1?

+1 for doing this, we're still on Bazel 5, planning to move to 6 in Q2 so we'll have to stick to this last release until then

**Review by thiagohmcruz — APPROVED** [review]
Thank you for keeping the Bazel 5 code path alive for a little longer!

**Comment by brentleyjones** [comment]
Merging so I can open up stacked PRs.

**Pull request merged**
